src/element/linearElementEditor.ts

```diff
@@ -157,40 +167,76 @@ export class LinearElementEditor {
       linearElementEditor.pointerDownState.lastClickedPoint
     ] as [number, number] | undefined;
     if (selectedPointsIndices && draggingPoint) {
-      const newDraggingPointPosition = LinearElementEditor.createPointAt(
-        element,
-        scenePointerX - linearElementEditor.pointerOffset.x,
-        scenePointerY - linearElementEditor.pointerOffset.y,
-        appState.gridSize,
-      );
+      if (
+        shouldRotateWithDiscreteAngle(event) &&
+        (selectedPointsIndices.length === 1 ||
+          prevSelectedPointsIndices?.length === 1) &&
+        element.points.length > 1
+      ) {
+        const prevSelectedIndex = prevSelectedPointsIndices?.[0];
+        const selectedIndex = selectedPointsIndices.filter(
+          (i) => i !== prevSelectedIndex,
+        )?.[0];
```

Comment:
not sure what the goal here is exactly with filter out previous selected points? EIther way, this will fail when you click on the same point twice. And since array access isn't type-safe (until we switch the TS compiler flag to make it safe), this will break.

Reply:
True, this part is not particularly careful, I'll have this rewritten.